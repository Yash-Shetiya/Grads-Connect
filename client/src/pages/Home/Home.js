import React, { useEffect, useState } from "react";
import logo from "../../assets/images/GradsConnectLogo.png";
import styles from "./Home.module.css";
import axios from "../../axios";

const Home = () => {
  const [testimonials, setTestimonials] = useState([]);

  useEffect(() => {
    axios
      .get(`http://localhost:8080/testimonials`)
      .then((res) => setTestimonials(res.data))
      .catch((err) => console.log(err));
  }, []);

  return (
    <div className={`container ${styles.homeContainer}`}>
      <img className={styles.logo} src={logo} alt="logo" />

      <div
        id="carouselExampleControls"
        className="carousel slide w-100"
        data-bs-ride="carousel"
      >
        <div className="carousel-inner">
          {testimonials.map((item, index) => {
            return (
              <div className={`carousel-item ${index === 0 ? "active" : ""}`}>
                <div>
                  <div className={styles.card}>
                    <div className={styles.cardThumb}>
                      <img src={item.src} alt="" className={styles.clientImg} />
                    </div>
                    <div className={styles.cardBody}>
                      <p className={styles.review}>{item.description}</p>
                      <div className={styles.rating}>
                        <span>- {item.name}</span>
                        <span className={styles.designation}>
                          {item.designation}
                        </span>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            );
          })}
        </div>
        <button
          className="carousel-control-next"
          type="button"
          data-bs-target="#carouselExampleControls"
          data-bs-slide="next"
        >
          <span className={styles.nextIcon}>&#8250;</span>
          <span className="visually-hidden">Next</span>
        </button>
      </div>
    </div>
  );
};

export default Home;
