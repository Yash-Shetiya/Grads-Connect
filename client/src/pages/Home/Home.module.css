.logo{
    display: block;
    margin-left: auto;
    margin-right: auto;
    width: 40%;
    margin-top: 2rem;
    margin-bottom: 1rem;
}

.homeContainer{
  min-height: 85vh;
  display: flex;
  flex-direction: column;
  justify-content: space-around;
}

.card{
    width: 80%;
    background: url('../../assets/images/Border.png');
    border-radius: 20px;
    display: flex;
    justify-content: space-around;
    align-items: center;
    padding: 20px;
    background-size:     cover;                      /* <------ */
    background-repeat:   no-repeat;
    margin: auto;
}
.cardThumb{
  width: 70%;
  padding: 10px 30px;
  margin: auto;
}


.clientImg{
  width: 90%;
  border-radius: 50%;
  margin: auto auto;
}

.cardBody{
  width:120%;
  margin-left: 10px;
  
}

.review{
  font-size: 20px;
  line-height: 30px;
  margin-top: 30px;
  font-family: 'Tahoma Regular', 'sans-serif';
}


.rating{
  display: flex;
  flex-direction: column;
  position:relative;
  bottom: 0;
  font-family: 'Tahoma Bold', sans-serif;
}
.designation{
  display: flex;
  justify-content: center;
  align-items: center;
 margin-right: 2rem;
  
}
span{
 margin-left: auto;
}
@media (max-width:800px) {
  
  .wrapper{
    width: 50%;
    margin: auto;
    min-width: 50%;
  }
  .card{
    flex-direction: column;
    background-image: none;
  }
  .cardThumb{
    width: 100%;
    border-radius: 10px;
  }
  .cardBody{
    width: 100%;
    min-height: auto;
    margin-left: 0;
  }
  
}


.nextIcon{
  font-size: 70px;
  color: #0570a3;
  right: 60px;
  position: relative;
}