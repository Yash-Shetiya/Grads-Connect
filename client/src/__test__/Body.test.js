import React from "react";
import Home from "../pages/Home/Home";
import OurGrafs from "../pages/OurGrads/OurGrads";
import { render } from "@testing-library/react";
import userEvent from "@testing-library/user-event";

describe("This suit is to test the Body component", () => {
  test("Snapshot of Body", () => {
    const { asFragment } = render(<Home title="hello world" />);
    expect(asFragment()).toMatchSnapshot();
  });

  test("finding title", () => {
    const { asFragment } = render(<OurGrafs year="2022" />);
    expect(asFragment()).toMatchSnapshot();
  });
});
