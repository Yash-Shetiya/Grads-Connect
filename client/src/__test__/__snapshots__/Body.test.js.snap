// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`This suit is to test the Body component Snapshot of Body 1`] = `
<DocumentFragment>
  <div
    class="container homeContainer"
  >
    <img
      alt="logo"
      class="logo"
      src="GradsConnectLogo.png"
    />
    <div
      class="carousel slide w-100"
      data-bs-ride="carousel"
      id="carouselExampleControls"
    >
      <div
        class="carousel-inner"
      />
      <button
        class="carousel-control-next"
        data-bs-slide="next"
        data-bs-target="#carouselExampleControls"
        type="button"
      >
        <span
          class="nextIcon"
        >
          â€º
        </span>
        <span
          class="visually-hidden"
        >
          Next
        </span>
      </button>
    </div>
  </div>
</DocumentFragment>
`;

exports[`This suit is to test the Body component finding title 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <span
      class="headerText"
    >
      Batch undefined
    </span>
    <div
      class="row"
    />
  </div>
</DocumentFragment>
`;
